prov/kubernetes/k8s_utils.go

```diff
@@ -241,3 +241,34 @@ func getNamespace(deploymentID string, objectMeta metav1.ObjectMeta) (string, bo
 
 	return namespace, isProvided
 }
+
+// Return the first healthy node found in the cluster
+func getHealthyNode(clientset kubernetes.Interface) (string, error) {
+	nodes, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to get nodes")
+	}
+	for _, node := range nodes.Items {
+		for _, cond := range node.Status.Conditions {
+			if cond.Type == corev1.NodeReady && cond.Status == corev1.ConditionTrue {
+				return node.ObjectMeta.Name, nil
+			}
+		}
+	}
+	//No healthy node found (error or warning to throw maybe)
```

Comment:
Indeed at least a warning should be printed.
So I suggest to return an error here and to emit a log event on use of getHealthyNode if no node could be retrieved.
We should not stop IMHO but simply generate an empty attribute or better just the port part.